HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -2842,30 +2747,253 @@ def self.create_curve_quint_linear(model, coeff, name)
     return curve
   end
 
+  def self.convert_net_to_gross_capacity(net_cap, watts_per_cfm, cfm_per_ton, fan_ratio, mode)
+    net_cap_ton = UnitConversions.convert(net_cap, 'Btu/hr', 'ton')
+    net_cap_watts = UnitConversions.convert(net_cap, 'Btu/hr', 'w')
+    cfm = net_cap_ton * cfm_per_ton
+    fan_power = watts_per_cfm * cfm * (fan_ratio**3)
+    if mode == :clg
+      gross_cap_watts = net_cap_watts + fan_power
+    else
+      gross_cap_watts = net_cap_watts - fan_power
+    end
+    gross_cap_btu_hr = UnitConversions.convert(gross_cap_watts, 'w', 'Btu/hr')
+    return gross_cap_btu_hr
+  end
+
+  def self.convert_net_to_gross_capacity_cop(net_cap, fan_power, mode, output_cop, net_cop = nil)
+    net_cap_watts = UnitConversions.convert(net_cap, 'Btu/hr', 'w')
+    if mode == :clg
+      gross_cap_watts = net_cap_watts + fan_power
+    else
+      gross_cap_watts = net_cap_watts - fan_power
+    end
+    if output_cop
+      net_power = net_cap_watts / net_cop
+      gross_power = net_power - fan_power
+      gross_cop = gross_cap_watts / gross_power
+      return gross_cop
+    else
+      gross_cap_btu_hr = UnitConversions.convert(gross_cap_watts, 'w', 'Btu/hr')
+      return gross_cap_btu_hr
+    end
+  end
```

Comment:
It's a bit weird to have a method that can return two different things depending on a boolean argument. How about have it return both values (though the second will be nil when `net_cop` is not provided)? You'd obviously have to update the calls to the method appropriately.

```suggestion
  def self.convert_net_to_gross_capacity_cop(net_cap, fan_power, mode, net_cop = nil)
    net_cap_watts = UnitConversions.convert(net_cap, 'Btu/hr', 'w')
    if mode == :clg
      gross_cap_watts = net_cap_watts + fan_power
    else
      gross_cap_watts = net_cap_watts - fan_power
    end
    if not net_cop.nil?
      net_power = net_cap_watts / net_cop
      gross_power = net_power - fan_power
      gross_cop = gross_cap_watts / gross_power
    end
    gross_cap_btu_hr = UnitConversions.convert(gross_cap_watts, 'w', 'Btu/hr')
    return gross_cap_btu_hr, gross_cop
  end
```